core/src/callback.jl

```diff
@@ -393,22 +398,131 @@ function set_fractional_flow_in_allocation!(
     return nothing
 end
 
-function set_control_params!(p::Parameters, node_id::NodeID, control_state::String)
-    node = getfield(p, p.graph[node_id].type)
-    idx = searchsortedfirst(node.node_id, node_id)
-    new_state = node.control_mapping[(node_id, control_state)]
+function update_parameters!(pump::Pump, parameter_update::ParameterUpdate)::Nothing
+    if parameter_update.node_type != NodeType.Pump
```

Comment:
This looks like a red flag to me, ideally we only call this with the correct data, so ParameterUpdate is too generic/uniony?